target/linux/realtek/files-5.10/arch/mips/rtl838x/setup.c

```diff
@@ -28,6 +29,46 @@
 
 extern struct rtl83xx_soc_info soc_info;
 
+void plat_get_frequencies(u32 *cpu, u32 *lxb, u32 *mem)
+{
+	struct device_node *np;
+	struct clk *clk;
+
+	*cpu = 500000000; // fallback
+	*lxb = 0;
+	*mem = 0;
+
+	clk = clk_get(NULL, "cpu_clk");
+
+	if (IS_ERR(clk)) {
+		pr_err("no clock driver, using DT.\n");
```

Comment:
Hmm, `pr_*` stuff was deprecated for ages, but you don't have a dev to use `dev_err` with ... I haven't seen yet where you call this function from; is it possible at all to pass a 'dev' a long? Also probably helps with the function below, where you pass NULL.

Granted, I haven't looked at clk stuff in years ...

Reply:
init stuff rewritten